cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/PluginInstantiator.java

```diff
@@ -114,6 +142,12 @@ public ClassLoader getArtifactClassLoader(ArtifactDescriptor artifactDescriptor)
     return (Class<T>) getArtifactClassLoader(artifactDescriptor).loadClass(pluginClass.getClassName());
   }
 
+  @SuppressWarnings("unchecked")
+  public <T> Class<T> loadClass(Location location,
```

Comment:
there should only be one way to load a class.  Do we still need the one that takes an ArtifactDescriptor?  Seems like we could use Location everywhere instead